### PR TITLE
feat: pod lifecycle filtering

### PR DESCRIPTION
**Pull request by pintohutch**
Only pods that are not "Failed" or "Succeeded" will be default candidates for scraping.

BREAKING CHANGE: this changes the default behavior of PodMonitoring and ClusterPodMonitoring

### CODE REVIEW

**Comment by pintohutch** [comment]
Fixes #145

**Comment by pintohutch** [comment]
The proposal here is similar to the prometheus-operator [approach](https://github.com/prometheus-operator/prometheus-operator/pull/5049), however it moves the boolean up to the pod-level, as opposed to the endpoint level. The thinking behind this is that since the [lifecycle phase](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) label is at the pod-level, then the configuration for scraping should be at the pod-level as well.

The drawback to this approach is: If one has a `PodMonitoring` with multiple scrape `endpoints`, and it's desired to have one endpoint be scraped regardless of pod status, while the other scraped only if the pod is "running", then this is not possible.

However, I can't imagine a case where this would be desired as, by definition, all containers in a `Succeeded` or `Failed` pod have exited. If there is some valid use case that we're not foreseeing here, it should still be possible by splitting up the `PodMonitoring` by scrape endpoint.

The considerations for making this a new default (as opposed to opt-in) are:
* Ignoring failed pods is a net beneficial change for managed collection and we want all users to have the feature with little friction.
* Scraping a "failed" pod should be the exception and not the rule. The majority of existing `PodMonitorings` shouldn't need to specify any new configuration to be covered.

Happy to discuss more though if there are other thoughts.

**Comment by pintohutch** [comment]
cc @lyanco @bwplotka for your input ^

File: pkg/operator/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -1016,6 +1038,10 @@ type ClusterPodMonitoringSpec struct {
 	TargetLabels TargetLabels `json:"targetLabels,omitempty"`
 	// Limits to apply at scrape time.
 	Limits *ScrapeLimits `json:"limits,omitempty"`
+	// FilterRunning will drop any pods that are in the "Failed" or "Succeeded"
```
Comment (bwplotka) [inline pkg/operator/apis/monitoring/v1/types.go:1041]: Can we add to the comment what's the benefit for enabling this for the user?

**Review by bwplotka — APPROVED** [review]
I'm fine with that 👍🏽 LGTM for pod level too.

QQ: What's the summary benefits of this? Just to reduce overhead? Decrease up metrics samples a bit?

File: pkg/operator/apis/monitoring/v1/types.go (hunk 1 above)
Comment (pintohutch) [reply pkg/operator/apis/monitoring/v1/types.go:1041]: Done.

**Comment by pintohutch** [comment]
@bwplotka - added context in the field description to answer your question. Basically the cases from #145 are the motivation. Helping folks reduce their monitoring bill is also a nice benefit 💸

File: pkg/operator/apis/monitoring/v1/types.go [hunk 2]
```diff
@@ -1019,6 +1041,14 @@ type ClusterPodMonitoringSpec struct {
...
+	// pod lifecycle.
+	// See: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+	// This prevents noisy errors when attempting to scrape Suceeded pods from
```
Comment (bwplotka) [inline pkg/operator/apis/monitoring/v1/types.go:1047]: It's not errors really, the succedded (completed) jobs which are not cleaned up report normal metrics just perhaps not relevant in majority of cases (if not all). (:

**Review by bwplotka — APPROVED** [review]
LGTM! 💪🏽 Thanks! Just minor comment nit.

File: pkg/operator/apis/monitoring/v1/types.go (hunk 2 above)
Comment (pintohutch) [reply pkg/operator/apis/monitoring/v1/types.go]: Done.

**Pull request merged**